src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -26,6 +27,58 @@ object WdlExpression {
 
   val unaryOperators = Set("LogicalNot", "UnaryPlus", "UnaryNegation")
 
+  val preEvaluableFunctions_SingleParameter: Seq[String] = Seq("read_int", "read_string")
+
+  /**
+   * Look within the expression for filenames which aren't explicitly listed as outputs. 
+   */
+  def preevaluateExpressionForFilenames(ast: AstNode, lookup: ScopedLookupFunction, functions: WdlFunctions): Try[Seq[WdlFile]] = {
+    ast match {
+      // This is the only case which actually pre-evaluates anything. The other cases are just buck-passers:
+      case a: Ast if a.getName == "FunctionCall"
```

Comment:
It'd be nice if you could boil the stuff between the `if` and the rocket into a function which took an `Ast` and returned true/false with a simple name which describes what the boolean represents